browser/ui/webui/brave_shields/shields_panel_data_handler.cc

```diff
@@ -67,52 +66,45 @@ void ShieldsPanelDataHandler::GetSiteSettings(
 }
 
 void ShieldsPanelDataHandler::SetAdBlockMode(AdBlockMode mode) {
-  auto* shields_data_ctrlr = GetActiveShieldsDataController();
-  DCHECK(shields_data_ctrlr);
+  DCHECK(observed_shields_data_ctrl_);
 
-  shields_data_ctrlr->SetAdBlockMode(mode);
+  observed_shields_data_ctrl_->SetAdBlockMode(mode);
 }
 
 void ShieldsPanelDataHandler::SetFingerprintMode(FingerprintMode mode) {
-  auto* shields_data_ctrlr = GetActiveShieldsDataController();
-  DCHECK(shields_data_ctrlr);
+  DCHECK(observed_shields_data_ctrl_);
 
-  shields_data_ctrlr->SetFingerprintMode(mode);
+  observed_shields_data_ctrl_->SetFingerprintMode(mode);
 }
 
 void ShieldsPanelDataHandler::SetCookieBlockMode(CookieBlockMode mode) {
-  auto* shields_data_ctrlr = GetActiveShieldsDataController();
-  DCHECK(shields_data_ctrlr);
+  DCHECK(observed_shields_data_ctrl_);
 
-  shields_data_ctrlr->SetCookieBlockMode(mode);
+  observed_shields_data_ctrl_->SetCookieBlockMode(mode);
 }
 
 void ShieldsPanelDataHandler::SetIsNoScriptsEnabled(bool is_enabled) {
-  auto* shields_data_ctrlr = GetActiveShieldsDataController();
-  DCHECK(shields_data_ctrlr);
+  DCHECK(observed_shields_data_ctrl_);
 
-  shields_data_ctrlr->SetIsNoScriptEnabled(is_enabled);
+  observed_shields_data_ctrl_->SetIsNoScriptEnabled(is_enabled);
 }
 
 void ShieldsPanelDataHandler::SetHTTPSEverywhereEnabled(bool is_enabled) {
-  auto* shields_data_ctrlr = GetActiveShieldsDataController();
-  DCHECK(shields_data_ctrlr);
+  DCHECK(observed_shields_data_ctrl_);
 
-  shields_data_ctrlr->SetIsHTTPSEverywhereEnabled(is_enabled);
+  observed_shields_data_ctrl_->SetIsHTTPSEverywhereEnabled(is_enabled);
 }
 
 void ShieldsPanelDataHandler::SetBraveShieldsEnabled(bool is_enabled) {
-  auto* shields_data_ctrlr = GetActiveShieldsDataController();
-  DCHECK(shields_data_ctrlr);
+  DCHECK(observed_shields_data_ctrl_);
```

Comment:
I don't think these should be DCHECKs, this can potentially be null if no tabs are active which might happen when closing the window I think

Reply:
yea, a null check here would suffice rather than dchecks